acme/acme/client.py

```diff
@@ -624,7 +624,32 @@ def _send_request(self, method, url, *args, **kwargs):
         kwargs.setdefault('headers', {})
         kwargs['headers'].setdefault('User-Agent', self.user_agent)
         kwargs.setdefault('timeout', self._default_timeout)
-        response = self.session.request(method, url, *args, **kwargs)
+        try:
+            response = self.session.request(method, url, *args, **kwargs)
+        except requests.exceptions.RequestException as e:
+            """ Regex for parsing error messages"""
+            """ Regular expression groups"""
+            """ Match Group #1: Attempted connection host"""
+            """ Match Group #2: urllib3 Error Number"""
+            """ Match Group #3: urllib3 Error String"""
+            """ Example:"""
+            """HTTPSConnectionPool(host='acme-v01.api.letsencrypt.org', 
+            port=443): Max retries exceeded with url: /directory 
+            (Caused by NewConnectionError('
+            <requests.packages.urllib3.connection.VerifiedHTTPSConnection
+            object at 0x108356c50>: Failed to establish a new connection:
+            [Errno 65] No route to host',))"""
+            """ Match Group #1: acme-v01.api.letsencrypt.org"""
+            """ Match Group #2: /directory"""
+            """ Match Group #3: [Errno 65]"""
+            """ Match Group #4: No route to host"""
+            error_string = str(e.message)
+            error_regex = r".*host='(\S*)'.*url\: (\/\w*).*(\[Errno \d+\])([A-Za-z ]*)"
+            host, directory, error_no, error_msg = re.match(error_regex, error_string).groups()
```

Comment:
We could also use "uri" rather than "path", in case any query strings come into play in the future.  I'll leave it as path for the time being like you're requesting.